Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Ride.hs

```diff
@@ -137,3 +164,109 @@ getRideStatus rideId personId = withLogTag ("personId-" <> personId.getId) do
         customer = SPerson.makePersonAPIEntity decRider tag,
         driverPosition = mbPos <&> (.currPoint)
       }
+
+editLocation ::
+  ( CacheFlow m r,
+    EncFlow m r,
+    EsqDBFlow m r,
+    MonadFlow m,
+    HasField "shortDurationRetryCfg" r RetryCfg,
+    HasFlowEnv m r '["nwAddress" ::: BaseUrl]
+  ) =>
+  Id SRide.Ride ->
+  (Id SPerson.Person, Id DM.Merchant) ->
+  EditLocationReq ->
+  m APISuccess
+editLocation rideId (_, merchantId) req = do
+  when (isNothing req.origin && isNothing req.destination) do
+    throwError PickupOrDropLocationNotFound
+
+  ride <- QRide.findById rideId >>= fromMaybeM (RideNotFound rideId.getId)
+
+  let attemptsLeft = fromMaybe 1 ride.allowedEditLocationAttempts
```

Comment:
In this way if ride.allowedEditLocationAttempts is Nothing, then customer can edit location infinite no if times.
Correct your logic.

Reply:
Please check line no 235 , attempts are getting updated so 1 will get updated to 0

Reply:
If value Nothing then default should come from same config as added in previous comment.